boards: arm: stm32_min_dev: Minimal enhancements

gon1332 opened the pull request:
Some minor enhancements will follow.

Signed-off-by: Ioannis Konstantelias <ikonstadel@gmail.com>

gon1332 committed:
boards: arm: stm32_min_dev: Enhance doc with pinout

Added the pinout diagram for the blue STM32 Minimum Development board.

Signed-off-by: Ioannis Konstantelias <ikonstadel@gmail.com>

dbkinder commented on boards/arm/stm32_min_dev/doc/index.rst:
```diff
@@ -43,6 +43,18 @@ Additional device support is left for the user to implement.
 More information on hooking up peripherals and lengthy how to articles can be
 found at `EmbedJournal`_.
 
+The pinout diagram of STM32 Minimum Deveopment Blue Pill board can be seen
```
s/Deveopment/Development/
s/bellow/below/

dbkinder reviewed (changes requested):
a couple of misspellings...

gon1332 committed:
boards: arm: stm32_min_dev: Remove broken links

Links showing the the variants of the board were broken. Now they are
gone.

Signed-off-by: Ioannis Konstantelias <ikonstadel@gmail.com>

gon1332 committed:
boards: arm: stm32_min_dev: Document connection info

Updated connection info and restructered a bit some paragraphs to stay
consistent with the structure of other boards.

Signed-off-by: Ioannis Konstantelias <ikonstadel@gmail.com>

gon1332 commented:
@dbkinder Do you want me to squash the commits?

dbkinder reviewed (approved):
OK for the docs, but curious why you removed the links to the blue/black pill boards? Oh, and yes, project convention would be to squash commits like these.

gon1332 commented:
> OK for the docs, but curious why you removed the links to the blue/black pill boards?

They are not working anymore. I tried to find something similar but with no success.
